packages/core/core-flows/src/order/workflows/exchange-request-item-return.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  OrderChangeDTO,
+  OrderDTO,
+  OrderExchangeDTO,
+  OrderWorkflow,
+  ReturnDTO,
+} from "@medusajs/types"
+import { ChangeActionType } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createStep,
+  createWorkflow,
+  transform,
+  when,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../common"
+import { createOrderChangeActionsStep } from "../steps/create-order-change-actions"
+import { createReturnsStep } from "../steps/create-returns"
+import { previewOrderChangeStep } from "../steps/preview-order-change"
+import { updateOrderExchangesStep } from "../steps/update-order-exchanges"
+import {
+  throwIfIsCancelled,
+  throwIfItemsDoesNotExistsInOrder,
+  throwIfOrderChangeIsNotActive,
+  throwIfOrderIsCancelled,
+} from "../utils/order-validation"
+
+const validationStep = createStep(
+  "request-item-return-validation",
+  async function ({
+    order,
+    orderChange,
+    orderReturn,
+    orderExchange,
+    items,
+  }: {
+    order: OrderDTO
+    orderReturn: ReturnDTO
+    orderExchange: OrderExchangeDTO
+    orderChange: OrderChangeDTO
+    items: OrderWorkflow.OrderExchangeRequestItemReturnWorkflowInput["items"]
+  }) {
+    throwIfOrderIsCancelled({ order })
+    throwIfIsCancelled(orderExchange, "Exchange")
+    throwIfIsCancelled(orderReturn, "Return")
+    throwIfOrderChangeIsNotActive({ orderChange })
+    throwIfItemsDoesNotExistsInOrder({ order, inputItems: items })
+  }
+)
+
+export const orderExchangeRequestItemReturnWorkflowId =
+  "exchange-request-item-return"
+export const orderExchangeRequestItemReturnWorkflow = createWorkflow(
+  orderExchangeRequestItemReturnWorkflowId,
+  function (
+    input: WorkflowData<OrderWorkflow.OrderExchangeRequestItemReturnWorkflowInput>
+  ): WorkflowData<OrderDTO> {
+    const orderExchange = useRemoteQueryStep({
+      entry_point: "order_exchange",
+      fields: ["id", "order_id", "return_id"],
+      variables: { id: input.exchange_id },
+      list: false,
+      throw_if_key_not_found: true,
+    }).config({ name: "exchange-query" })
+
+    const existingOrderReturn = when({ orderExchange }, ({ orderExchange }) => {
+      return orderExchange.return_id
+    }).then(() => {
+      return useRemoteQueryStep({
+        entry_point: "return",
+        fields: ["id", "status", "order_id"],
+        variables: { id: orderExchange.return_id },
+        list: false,
+        throw_if_key_not_found: true,
+      }).config({ name: "return-query" }) as ReturnDTO
+    })
+
+    const createdReturn = when({ orderExchange }, ({ orderExchange }) => {
+      return !orderExchange.return_id
```

Comment:
thought(non-blocking): Don't we want the return to always exist at this point? This is requesting to return an item. It seems a bit off to always create the return if it doesn't exist. Same with claims.

Wdyt?

Reply:
But at which point we are creating a new request for the exchange or the claim?
I thought this would be created implicitly when request the return of an item in the claim or the exchange.
Will it be a separated request just for that?

Reply:
I think it will be separate requests. So a full exchange lifecycle would look something like the following:
```
POST /admin/exchanges -> initiate change

POST /admin/exchanges/:id/inbound-items -> request return of item one
POST /admin/exchanges/:id/inbound-items -> request return of item two

POST /admin/exchanges/:id/outbound-items -> request addition of item three
POST /admin/exchanges/:id/outbound-items -> request addition of item four

POST /admin/exchanges/:id/inbound-shipping -> add return shipping for returning items
POST /admin/exchanges/:id/outbound-shipping -> add shipping for new items

POST /admin/exchanges/:id/confirm -> confirm the change request
```

Don't mind the API design here, but the overall flow would be something like this.

Let me know if this makes sense.

Reply:
Ahh wait, I think I understand what you mean now. I believe we should create the return as part of initiating the exchange request. So on first request:
```
POST /admin/exchanges -> initiate change -> create exchange + return
```

As you can't really have an exchange without a return right?